libs/ai-endpoints/langchain_nvidia_ai_endpoints/reranking.py

```diff
@@ -60,125 +62,44 @@ def __init__(self, **kwargs: Any):
             environment variable.
         """
         super().__init__(**kwargs)
+        infer_path = "{base_url}/ranking"
+        # not all models are on https://integrate.api.nvidia.com/v1,
+        # those that are not are served from their own endpoints
+        if model := determine_model(self.model):
+            if model.endpoint:  # some models have custom endpoints
+                infer_path = model.endpoint
```

Comment:
you are quite correct. this is code that's duplicated across all connectors and should move into _NVIDIAClient.

i ran into an issue w/ differing types when making the _NVIDIAClient.model as Model instead of a str.

let me follow up this commit w/ cleanup of the Model/str handling.